programs/{{project-name}}/src/lib.rs

```diff
@@ -46,19 +46,20 @@ pub mod {{rust-name}} {
         let pool_type = [0u8; 32];
         let mut program_id_hash = hash(&ctx.program_id.to_bytes()).to_bytes();
         program_id_hash[0] = 0;
-        let checked_inputs = [
+        let mut checked_inputs = [
```

Comment:
I think it would be better to do:

```rust
let mut checked_inputs: [[u8; 32]; NR_CHECKED_INPUTS] = [[0u8; 32]; NR_CHECKED_INPUTS];
checked_inputs[0] = program_id_hash;
checked_inputs[1] = inputs_des.transaction_hash;
```

This way we allocate memory only once on stack, for the size known from start (`NR_CHECKED_INPUTS`), instead of creating a Vector and pushing to it.

Reply:
agreed this is better